### ISSUE TITLE
Ec2MultiRegionSnitch should not enforce a public RPC address

### ISSUE THREAD

**Issue by fee-mendes** [scylladb/scylla#10236]
*Installation details*
Scylla version (or git commit hash): `4.6.0-0.20220306.2f2d22a86`
Cluster size: 2x i3.xlarge
OS (RHEL/CentOS/Ubuntu/AWS AMI): ` ami-0607f8830ea1f5981`

Scylla - unlike Cassandra - seems to still be enforcing `broadcast_rpc_address` to a public IP rather than parsing what the user has choosen, as shown in [ec2_multi_region_snitch.cc](https://github.com/scylladb/scylla/blob/master/locator/ec2_multi_region_snitch.cc#L64). This appears to have been introduced in https://github.com/scylladb/scylla/commit/c462af5549c9197139b49bc882fd06affef1de66 with the claim that:

```
            // Cassandra 2.1 manual explicitly instructs to set broadcast_address
            // value to a public address in cassandra.yaml.
```

This comment don't make too much sense, because:
- `broadcast_address` has no relation to `broadcast_rpc_address`
- Cassandra documentation has no mention of `broadcast_rpc_address` being required to be set to a public IP address
- Cassandra 4.0.3 **allows** the user to set `broadcast_rpc_address` to a public IP, but defaults to private by default

# How to reproduce
1 - Create a 2-node Scylla cluster w/ Ec2MultiRegionSnitch
2 - Check `system.peers` table:

```cql
cqlsh> select peer,data_center,host_id,preferred_ip,rack,rpc_address,schema_version from system.peers;

 peer          | data_center | host_id                              | preferred_ip  | rack | rpc_address   | schema_version
---------------+-------------+--------------------------------------+---------------+------+---------------+--------------------------------------
 18.216.98.219 |   us-east-2 | d9443741-a12e-4bbb-91ce-9931cece589c | 172.31.43.122 |   2c | 18.216.98.219 | 95c3fca5-c463-3aba-98c6-1c0b3fac5b58

(1 rows)
cqlsh> 
```

### Cassandra
```cql
cqlsh> SELECT peer,data_center,host_id,preferred_ip,rack,rpc_address,schema_version from system.peers;

 peer          | data_center | host_id                              | preferred_ip  | rack       | rpc_address   | schema_version
---------------+-------------+--------------------------------------+---------------+------------+---------------+--------------------------------------
 52.15.104.255 |   us-east-2 | 42c0b717-775f-4998-a420-0388fe8b4e70 | 172.31.42.126 | us-east-2c | 172.31.42.126 | 2207c2a9-f598-3971-986b-2926e09e239d

(1 rows)
cqlsh> 
```

### Config difference
```shell
cassandra.yaml:rpc_address: 0.0.0.0
cassandra.yaml:broadcast_rpc_address: 172.31.42.126
/etc/scylla/scylla.yaml:broadcast_rpc_address: 172.31.42.126
/etc/scylla/scylla.yaml:rpc_address: 0.0.0.0
```

**Comment by fee-mendes** [comment]
@slivne ping

**Comment by fee-mendes** [comment]
Actually after taking a closer look this was introduced when we introduced the `broadcast_rpc_address` at https://github.com/scylladb/scylla/commit/7ddf8963f598edc8b3dc13ee5cadd696eb021910 :-) 

@asias do you know why it was decided to enforce a public RPC address back then for this snitch?

**Comment by asias** [comment]
Yes, it makes sense to  set broadcast_rpc_address in Ec2MultiRegionSnitch only when user does not specifies broadcast_rpc_address. If user sets broadcast_rpc_address, we should keep using it.

**Comment by asias** [comment]
PR sent: https://github.com/scylladb/scylla/pull/10519

**Comment by avikivity** [comment]
@fee-mendes how likely is this to be encountered by a user? I'm evaluating whether to backport or not.

**Comment by fee-mendes** [comment]
@avikivity its been like this since when the `broadcast_rpc_address` option was introduced, so this isn't likely to be encountered soon. For single DC deployments users can still rely on Ec2Snitch for the time being, for MultiDC they can peer among the regions and still use Ec2Snitch OR fallback to GossipingPropertyFileSnitch, so this ain't a blocker.

**Comment by avikivity** [comment]
Thanks, not backporting per above.
